### PR TITLE
updates deal.II stable version to 9.3.0 and adds early deprecation

### PR DESCRIPTION
**Pull request by koecher**
updates deal.II stable version to 9.3.0 and adds early deprecation warnings.

I've tested a run on centos 7 with only the deal.II package, looks good from my side.

@tjhei we should introduce a new branch for deal.II v9.3

### CODE REVIEW

**Comment by gfcas** [comment]
@koecher @tjhei I also updated my local candi version and tested it under WSL Ubuntu 20.04 with nearly every package. It worked, however, only when using the bugfix of pr #159.

**Comment by tjhei** [comment]
I will work on the branch in the next few days.

File: deal.II-toolchain/packages/dealii.package [hunk 1]
```diff
@@ -17,6 +17,7 @@ INSTALL_PATH=${ORIG_INSTALL_PATH}/deal.II-${VERSION}
 
 CONFOPTS=" \
 -D CMAKE_BUILD_TYPE=DebugRelease \
+-D DEAL_II_EARLY_DEPRECATIONS:BOOL=ON \
```
Comment (tjhei) [inline deal.II-toolchain/packages/dealii.package:20]: I am not sure if we should set this by default. This is more of a developer option.

**Comment by tjhei** [comment]
@gfcas how does #159 manifest itself for you? Code should be compiling without the patch, right?

**Comment by gfcas** [comment]
The CI on ubuntu 18.04 and 20.04 show exactly the same problem.

**Comment by koecher** [comment]
@tjhei okay, I've set the default to OFF, I thought it would be good to know as soon as possible for deprecations.

**Comment by tjhei** [comment]
Ahh, thanks @gfcas . This is also a bug in deal.ii...

**Comment by gfcas** [comment]
As I understood, this might be even a cmake bug dealii/#11478.

File: deal.II-toolchain/packages/dealii.package [hunk 2]
```diff
@@ -17,6 +17,7 @@ INSTALL_PATH=${ORIG_INSTALL_PATH}/deal.II-${VERSION}
 
 CONFOPTS=" \
 -D CMAKE_BUILD_TYPE=DebugRelease \
+-D DEAL_II_EARLY_DEPRECATIONS:BOOL=OFF \
 -D DEAL_II_WITH_MPI:BOOL=ON \
```
Comment (gfcas) [inline deal.II-toolchain/packages/dealii.package:21]: @tjhei as this is a developer option, maybe we should enable this feature if DEVELOPER_MODE=ON?

Comment (koecher) [reply deal.II-toolchain/packages/dealii.package:21]: or putting it to the "global" DEAL_II_CONFOPTS in candi.cfg would be another idea.

**Comment by koecher** [comment]
maybe we should let this option to developers only, they can add them in the globally known DEAL_II_CONFOPTS for instance to the configuration

**Pull request closed without merging**